workflow/context_propagator.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package workflow
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/cadence/internal"
+)
+
+type (
+	// HeaderReader is an interface to read information from cadence headers
+	HeaderReader = internal.HeaderReader
+
+	// HeaderWriter is an interface to write information to cadence headers
+	HeaderWriter = internal.HeaderWriter
+
+	// ContextPropagator is an interface that determines what information from
+	// context to pass along
+	ContextPropagator = internal.ContextPropagator
+)
+
+// ContextKey to be used for the StringMapPropagator
+type ContextKey string
+
+// stringMapPropagator propagates the list of keys across a workflow,
+// interpreting the payloads as strings. This is provided as a sample for other
```

Comment:
Why not interpreting payloads as bytes?

Reply:
It's just a sample. The tracing propagator already has a sample of marshalling payloads into bytes, so wanted this sample to deal with strings..

Reply:
I'm not sure that sample should be part of the core library. May be move it into a samples repo?

Reply:
Would you prefer if I leave it as internal and use it just for test purposes?

Reply:
I'll move this to be a local struct that is used only by my test..